apps/server/src/modules/tldraw/controller/tldraw.ws.ts

```diff
@@ -93,23 +60,67 @@ export class TldrawWs implements OnGatewayInit, OnGatewayConnection {
 		return parsedCookies;
 	}
 
-	private closeClientAndLogError(client: WebSocket, code: WsCloseCodeEnum, data: string, err: unknown): void {
-		client.close(code, data);
-		this.logger.warning(new WebsocketCloseErrorLoggable(err, `(${code}) ${data}`));
-	}
-
 	private async authorizeConnection(drawingName: string, token: string): Promise<void> {
 		if (!token) {
 			throw new UnauthorizedException('Token was not given');
 		}
 
-		await firstValueFrom(
-			this.httpService.get(`${this.apiHostUrl}/v3/elements/${drawingName}/permission`, {
-				headers: {
-					Accept: 'Application/json',
-					Authorization: `Bearer ${token}`,
-				},
-			})
+		try {
+			const apiHostUrl = this.configService.get<string>('API_HOST');
+			await firstValueFrom(
+				this.httpService.get(`${apiHostUrl}/v3/elements/${drawingName}/permission`, {
+					headers: {
+						Accept: 'Application/json',
+						Authorization: `Bearer ${token}`,
+					},
+				})
+			);
+		} catch (err) {
+			if (isAxiosError(err)) {
+				switch (err.response?.status) {
+					case 400:
+					case 404:
+						throw new NotFoundException();
+					case 401:
+					case 403:
+						throw new UnauthorizedException();
+					default:
+						throw new InternalServerErrorException();
+				}
```

Comment:
I think you need to throw WsException
https://docs.nestjs.com/websockets/exception-filters

Reply:
We've tested it and WsException wasn't caught in your GlobalFilter anyway so I'd leave it like that for now.

But this is definitely something to consider in the future. We would need to create our own exceptions from WsException as there is nothing like WsNotFoundException etc.